src/App.js

```diff
@@ -0,0 +1,78 @@
+import "./App.css";
+import { useEffect, useState } from "react";
+import { listMealsByFirstLetter, getMealById } from './modules/recipies/recipiesService';
+import { SearchBar } from "./modules/recipies/SearchBar";
+import { RecipeList } from "./modules/recipies/RecipeList";
+
+
+function App() {
+  const [recipes, setRecipes] = useState([]);
+  const [selectedRecipe, setSelectedRecipe] = useState(null);
+
+  const handleSearch = (letter) => {
+    listMealsByFirstLetter(letter)
+      .then((recipes) => {
+        setRecipes(recipes || []);
+        setSelectedRecipe(null);
+      })
+      .catch((error) => console.error("Error fetching recipes:", error));
+  };
+
+  const handleRecipeClick = (id) => {
+    getMealById(id)
+      .then((recipe) => {
+        setSelectedRecipe(recipe);
+        setRecipes([]);
+      })
+      .catch((error) => console.error("Error fetching recipe detail:", error));
+  };
+  const handleBackToList = () => {
+    setSelectedRecipe(null);
+    handleSearch('a');
+  };
+
+  useEffect(() => {
```

Comment:
I don't understand login in the useEffect hook. First, when user enters the page, we load recipes. Then we clear them out every 20 seconds. This does not seem to be a good user experience. 

For now on this page let's remove useEffect completely and only have recipes available through search. Later, when the application progresses you can add useEffect hook to the RecipeList component to fetch the recipes on the main page.